src/ol/render/canvas.js

```diff
@@ -324,6 +324,22 @@ export function measureTextWidth(font, text) {
 }
 
 
+/**
+ * Measure text width using a cache.
+ * @param {string} text The text to measure.
+ * @param {string} font The font.
+ * @param {Object<string, number>} cache A lookup of cached widths by text.
+ * @returns {number} The text width.
+ */
+export function measureAndCacheTextWidth(text, font, cache) {
```

Comment:
While adding and editing comments on your PR, I decided to juggle the order of these args to better match the `measureTextWidth` signature.  So the reason things aren't working here is that this function is called with `font, text, cache`.

As mentioned above, I pushed a version of the commit I had in mind to #8988.  Tests are passing there.